### ISSUE TITLE
[BUG] Disabled filter affecting Dashboard results

### ISSUE THREAD

**Issue by joshuali925** [opensearch-project/OpenSearch-Dashboards#4744]
**Describe the bug**

In dashboard there can be global filters applied to all visualizations, and local filters defined inside the visualizations and only applies to the visualization. The global filter will override the local filter if there's a conflict, even when it is disabled. Seeing different results with a disabled filter being present/absent is confusing.

**To Reproduce**
1. create a metrics visualization on flights sample data
2. add a filter on `cancelled` to be `true`
3. check results (41 in my case)
4. create a dashboard and add the visualization
5. add a global filter `cancelled` is `true`, result is the same
6. disable the global filter, result increases (322 in my case)
7. remove the global filter, result changes (back to 41)

**Expected behavior**
removing a disabled filter should not modify results

**OpenSearch Version**
main branch

**Dashboards Version**
main branch

**Plugins**
None

**Screenshots**

**Host/Environment (please complete the following information):**
 - OS: [e.g. iOS] mac
 - Browser and version [e.g. 22] chrome

**Additional context**

I understand this might be expected behavior to have global filter (including disabled) to override local ones, but it is confusing in this case. I think the better solution is to show the local filters in each visualization, and show whether they are being used or being overridden.

**Comment by rosepark4861** [comment]
This is very confusing behavior.  The normal expectation would be that when a filter is disabled that the report will show the equivalent results to what would be shown if the filter is deleted.
Is it intentional behavior that you want disabling a dashboard level filter to override/disable any existing discover or visualization filter on the same field, or similarly that disabling a visualization filter means you want it to override/disable any existing discover filter on the same field?  
I certainly didn't expect this behavior and don't know if there is significant value in the way this is working.
There is value in being able to disable a filter without deleting it, as it is easier to reenable an existing filter than it is to create the filter again.  My expectation would have been that disabling a filter should result in the same results being displayed on the report as what would be displayed on the report when the filter is deleted.
If both of these behaviors have value then I would suggest that there needs to be two separate disable options shown to the user, allowing the user to either 1) disable the filter they are modifying (meaning to show the report data as if this filter had been deleted), or 2) disable all filters (dashboard, visualization, discover) that are filtering on this value.
Providing separate selectable options for both possible actions would help clear up the current unexpected behavior when disabling a filter as the user would realize when choosing selection 2 their selection will impact the report data because the filter isn't being disabled itself, but is now becoming a new overriding rule to disable filters at other levels .
If you don't want to provide two different disable options then I would suggest that the better behavior is that a disabled filter should cause the report to show the same data that is displayed when the filter is removed.  The current behavior where disabling the filter vs. deleting the filter can result in significantly different results being shown in the report is very unexpected and could easily lead to users making incorrect conclusions and possibly taking incorrect actions when examining the resulting report data.

**Comment by ashwin-pc** [comment]
@joshuali925 yes this is a weird UX we inherited. @kgcreative I had called this out earlier as one of the weird behavior's I noticed while developing VisBuilder too. Is this something we are looking to fix sometime soon? cc: @mnkugler @wbeckler

**Comment by rosepark4861** [comment]
Not supporting an option to actually disable a filter (meaning ignore the filter and display the results as if the filter had been deleted) seems like a big functionality oversight.  What makes this an even bigger issue is that there is a disable option shown in the UI which people will assume does provide the functionality to disable the filter (meaning ignore it).
If the current behavior has some value then it needs a different name.  Perhaps something like "Cascade Disable", "Ripple Disable", "Overriding Disable", "Inherited Disable", or "Propagated Disable" so that users of OpenSearch are not mislead into thinking this is actually a disable option the will really disable the filter as would be expected.
For me the expected disable functionality is the one that I want.  A disable option that shuts off and ignores the filter.

**Comment by wbeckler** [comment]
I think this can be considered a bug, and if someone were to change things to what is expected, nobody would say it was a regression of any sort.

**Comment by kgcreative** [comment]
I agree. It would be good to fix this so it behaves as expected.

**Comment by seraphjiang** [comment]
Thank you for reporting this issue. This issue has been open for a long time without recent activity, and OpenSearch Dashboards has since moved to the 3.x release line (currently 3.6, with 3.7 upcoming). The last supported 2.x release is 2.19.x.

We're closing this as part of a housekeeping effort to keep the backlog actionable. If this issue is still reproducible on a supported version (2.19.x or 3.x), please feel free to reopen this issue or file a new one with updated reproduction steps and version details.

We appreciate your contribution to OpenSearch!
